test/spec/Control/Biegunka/Source/GitSpec.hs

```diff
@@ -0,0 +1,121 @@
+module Control.Biegunka.Source.GitSpec (spec) where
+import           Test.Hspec
+import           Test.Hspec.Expectations.Contrib
+
+import           Control.Biegunka
+import           Control.Biegunka.Source.Git (git_ {-, git', failIfAhead-})
+import           Control.Lens
+import           Control.Monad               (liftM)
+import Control.Monad (void)
+import           Data.Maybe                  (listToMaybe)
+import           System.FilePath             ((</>))
+import           System.IO.Silently          (silence)
+import           System.IO.Temp              (withSystemTempDirectory)
+import qualified System.Process              as P
+import           System.Random
+
+spec :: Spec
+spec = do
+  describe "git_" $ around withBiegunkaDirectory $ do
+
+    context "when local path doesn't exist" $
+
+      it "creates new directory and set branch correctly" $ \tmp -> do
+        let repo = tmp </> "biegunka-test"
+        silence $ biegunka (set runRoot tmp . set biegunkaRoot (tmp </> ".biegunka")) run $
+          (retries 0 $ git_ testRepo "biegunka-test")
+        currentBranch repo `shouldReturn` Just "master"
+        haveCleanState repo `shouldReturn` True
```

Comment:
It's unclear to me what's the clean state is. Does it care whether there are untracked files in the working tree?